plugin/trino-hive/src/main/java/io/trino/plugin/hive/parquet/ParquetFileWriterFactory.java

```diff
@@ -101,17 +104,20 @@ public Optional<FileWriter> createFileWriter(
             return Optional.empty();
         }
 
+        Set<String> bloomFilterColumns = getParquetBloomFilterColumns(schema);
         ParquetWriterOptions parquetWriterOptions = ParquetWriterOptions.builder()
                 .setMaxPageSize(HiveSessionProperties.getParquetWriterPageSize(session))
                 .setMaxPageValueCount(HiveSessionProperties.getParquetWriterPageValueCount(session))
                 .setMaxBlockSize(HiveSessionProperties.getParquetWriterBlockSize(session))
                 .setBatchSize(HiveSessionProperties.getParquetBatchSize(session))
+                .setBloomFilterColumns(bloomFilterColumns)
                 .build();
 
         List<String> fileColumnNames = getColumnNames(schema);
         List<Type> fileColumnTypes = getColumnTypes(schema).stream()
                 .map(hiveType -> hiveType.getType(typeManager, getTimestampPrecision(session)))
                 .collect(toList());
+        validateBloomFilterColumns(fileColumnNames, fileColumnTypes, bloomFilterColumns);
```

Comment:
This validation should happen in ConnectorMetadata as this should be checked sooner in the coordinator rather than getting to creation of file writers on the workers.

Reply:
For now the validation is removed, as we would need to produce a map of dot-path to type in HiveMetadata. To support Bloom filters for columns with structural types, I've also removed the validation that a valid column is specified. For instance, `a row(b integer)`, specified as `a.b`. We do also support writing bloom filters for `a array<integer>`, specified as `a.list.element`, and `a map<integer, integer>` specified as `a.key_value.key` and `a.key_value.value`, but it does not look like we make use of Bloom filters when reading columns with array or map type. It does not look like we currently support writing Bloom filters for structural types with ORC.

Reply:
Would Hive or Spark make use of bloom filter if they are written for leaf elements of structured types ? We can write for those cases if other engines can utilize this on read today, otherwise we should skip. I think we should have validation rejecting struct type (if that's something we don't support) and types not in SUPPORTED_BLOOM_FILTER_TYPES in HiveMetadata. I think the syntax and validation for nested types should look similar to https://github.com/trinodb/trino/pull/15712 
We also need tests coverage for nested types.

Reply:
I believe engines which use parquet-mr will make use of Bloom filters of leaf elements of structured types.

To keep this simple to begin with, we can follow the same validation logic as for ORC and disallow writing Bloom filters on structured types, and handle the structured types in a follow up?

Reply:
ya we can handle structured types in a follow up